### PR TITLE
Update theming-plugin to always generate json file in same location

### PR DESCRIPTION
**Pull request by bjankord**
### Summary
This update resolves an issue where users would change directories into one of the component directories and run nightwatch tests. This would trigger webpack to run, hitting the theming-plugin and generating a `themeable-variables.json` file inside of the component directory.

After this update, when users change directories into one of the component directories and run nightwatch tests or run `npm start` from the root or `terra-site` directory, the `themeable-variables.json` file will always be output to `packages/terra-site`.

Thanks for contributing to Terra. 
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

File: packages/terra-site/tests/jest/theming-plugin-spec.test.jsx [hunk 1]
```diff
@@ -54,9 +54,8 @@ describe('theming-plugin', () => {
         },
       };
       rootNode.source = { input: { file: 'TestingComponent.scss' } };
-
       expect(testingFunction(rootNode)).toEqual(JSON.stringify(expectedHash));
-      expect(outputtedFileName).toEqual('themeable-variables.json');
+      expect(outputtedFileName.slice(outputtedFileName.length - 55)).toEqual('terra-core/packages/terra-site/themeable-variables.json');
```
Comment (emilyrohrbough) [inline packages/terra-site/tests/jest/theming-plugin-spec.test.jsx]: What does -55 do?

File: packages/terra-site/theming-plugin.js [hunk 2]
```diff
@@ -33,6 +34,6 @@ module.exports = postcss.plugin('theming-plugin', () => {
       }
     });
 
-    return fs.writeFileSync('themeable-variables.json', JSON.stringify(sortHash(variables)), 'utf8');
+    return fs.writeFileSync(path.resolve(__dirname, 'themeable-variables.json'), JSON.stringify(sortHash(variables)), 'utf8');
```
Comment (mjhenkes) [inline packages/terra-site/theming-plugin.js]: Could we pretty print the json?
`JSON.stringify(sortHash(variables)), 'utf8', 2)`

File: packages/terra-site/tests/jest/theming-plugin-spec.test.jsx (hunk 1 above)
Comment (bjankord) [reply packages/terra-site/tests/jest/theming-plugin-spec.test.jsx]: Wanted to grab the end of the file / path, or the last 55 characters. Using this to check that its within the `terra-site` folder. This slice returns `terra-core/packages/terra-site/themeable-variables.json`. Depending on where this code is run, your machine, my machine, travis, the outputtedFileName's file path is different. Slicing the string to just the path thats consistent gives us the info we need when run in various environments. Also, its a nod to not driving slow.

File: packages/terra-site/theming-plugin.js (hunk 2 above)
Comment (bjankord) [reply packages/terra-site/theming-plugin.js]: Yeah, good idea. That will make diffs of this file easier to reason about.

Comment (bjankord) [reply packages/terra-site/theming-plugin.js]: Updated: https://github.com/cerner/terra-core/pull/811/commits/205672d5bd4d59163e3899587fee1cfded292e07

File: packages/terra-site/themeable-variables.json [hunk 3]
```diff
@@ -1 +1,278 @@
...
+    "--terra-popup-content-close-font-size": "1.45rem",
+    "--terra-popup-content-close-font-weight": "bold",
+    "--terra-popup-content-close-line-height": "0",
+    "--terra-popup-content-close-margin-left": "0.3571rem",
+    "--terra-popup-content-close-margin-right": "0",
+    "--terra-popup-content-close-padding": "4px",
+    "--terra-popup-content-close-hover-color": "#999",
+    "--terra-popup-content-inner-background-color": "#fff",
+    "--terra-popup-content-inner-border": "1px solid #b2b5b6",
+    "--terra-popup-content-inner-box-shadow, 0 4px 8px 0 rgba(0, 0, 0, 0.35), 0 4px 3px -3px rgba(0, 0, 0": "0.5)",
```
Comment (bjankord) [inline packages/terra-site/themeable-variables.json]: Logged an issue about this malformatting: https://github.com/cerner/terra-core/issues/818
It should not impact users much as I don't think anyone is using this file other than us to generate the list of themable components here on https://terra-ui.herokuapp.com/getting-started#theming. For that we only use the top level keys.

File: packages/terra-site/themeable-variables.json [hunk 4]
```diff
@@ -1 +1,278 @@
...
+    "--terra-badge-color-positive": "#fff",
+    "--terra-badge-background-color-negative": "#bc0203",
+    "--terra-badge-color-negative": "#fff",
+    "--terra-badge-background-color-warning": "#da3b03",
+    "--terra-badge-color-warning": "#fff",
+    "--terra-badge-background-color-info": "#004c76",
+    "--terra-badge-color-info": "#fff"
+  },
+  "Base": {
+    "--terra-base-font-family, \"Helvetica Neue\", Helvetica, Arial": "sans-serif",
```
Comment (bjankord) [inline packages/terra-site/themeable-variables.json]: Logged an issue about this malformatting: #818

**Comment by bjankord** [comment]
I need to talk with @JakeLaCombe tomorrow about the utf-8 setting. I'm seeing the JSON files saved as `us-ascii` prior to this change and after the change, and this is what the jest tests are failing on.

File: packages/terra-site/theming-plugin.js [hunk 5]
```diff
@@ -33,6 +33,6 @@ module.exports = postcss.plugin('theming-plugin', () => {
         variables[component][matches[1]] = matches[2];
       }
     });
-    return fs.writeFileSync(path.resolve(__dirname, 'themeable-variables.json'), JSON.stringify(sortHash(variables), 'utf8', 2));
+    return fs.writeFileSync(path.resolve(__dirname, 'themeable-variables.json'), JSON.stringify(sortHash(variables), null, 2), 'utf8');
```
Comment (mjhenkes) [inline packages/terra-site/theming-plugin.js]: writeFileSync should default to utf8, but it doesn't hurt to specify.

**Pull request merged**